### PR TITLE
Document ProcessGroup

### PR DESCRIPTION
**Pull request by kshyatt**
This is exported but has no documentation.

### CODE REVIEW

File: base/multi.jl [hunk 1]
```diff
@@ -393,6 +393,17 @@ let next_pid = 2    # 1 is reserved for the client (always)
...
+    ProcessGroup(name::AbstractString, workers::Array{Any,1}, refs::Dict, topology::Symbol)
+
+A group of `workers` with some `topology`, named `name`. The group may be assigned work
+to do using constructs like `pmap`. There are three options for `topology`:
+
+* `:all_to_all` - all workers connect to all workers
+* `:master_slave` - the head worker connects to all the others, but they do not connect to each other
+* `:custom` - some custom `topology` which is specified using a [`ClusterManager`](:class:`ClusterManager`)
+"""
+ProcessGroup(name::AbstractString, workers::Array{Any,1}, refs::Dict, topology::Symbol)
```
Comment (tkelman) [inline base/multi.jl]: I don't think the signature should be necessary for documenting a type?

**Comment by amitmurthy** [comment]
Like `LocalProcess` this too harkens back to being exported from the initial distributed implementation. Both should be unexported as we don't expect users to use these types directly. At least, not yet. 

We can keep the doc string but remove it from the rst.

**Comment by tkelman** [comment]
> We can keep the doc string

were you going to move it to the other PR?

**Comment by kshyatt** [comment]
>were you going to move it to the other PR?

Yes.

**Pull request closed without merging**